platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/testapp/activity/BaseTest.java

```diff
@@ -1,116 +1,113 @@
+  private static final int MAX_TIMEOUT_FAILURE = 3;
+  private static int timeoutCount;
 
   @Rule
-  public ActivityTestRule<Activity> rule = new ActivityTestRule<>(getActivityClass());
+  public ActivityTestRule rule = new ActivityTestRule<>(getActivityClass());
 
   @Rule
-  public TestName testNameRule = new TestName();
+  public TestName testName = new TestName();
 
   protected MapboxMap mapboxMap;
-  protected MapboxIdlingResource idlingResource;
+  protected MapView mapView;
+  private final CountDownLatch latch = new CountDownLatch(1);
 
   @Before
   public void beforeTest() {
-    try {
-      Timber.e(String.format(
-        "%s - %s - %s",
-        getClass().getSimpleName(),
-        testNameRule.getMethodName(),
-        "@Before test: register idle resource"
-      ));
-      idlingResource = (MapboxIdlingResource) generateIdlingResource();
-      IdlingRegistry.getInstance().register(idlingResource);
-      Espresso.onIdle();
-      mapboxMap = idlingResource.getMapboxMap();
-    } catch (IdlingResourceTimeoutException idlingResourceTimeoutException) {
-      throw new RuntimeException(
-        String.format(
-          "Could not start %s test for %s.",
-          testNameRule.getMethodName(),
-          getActivityClass().getSimpleName()
-        )
-      );
-    }
+    initialiseMap();
+    holdTestRunnerForStyleLoad();
   }
 
-  protected IdlingResource generateIdlingResource() {
-    return new FinishLoadingStyleIdlingResource(rule.getActivity());
+  private void initialiseMap() {
+    try {
+      rule.runOnUiThread(() -> {
+        mapView = rule.getActivity().findViewById(R.id.mapView);
+        mapView.addOnDidFinishLoadingStyleListener(this);
+        mapView.getMapAsync(this::initMap);
+      });
+    } catch (Throwable throwable) {
+      throwable.printStackTrace();
+    }
   }
 
-  protected void validateTestSetup() {
-    if (!Mapbox.isConnected()) {
-      Timber.e("Not connected to the internet while running test");
+  private void holdTestRunnerForStyleLoad() {
+    boolean interrupted;
+    try {
+      interrupted = latch.await(WAIT_TIMEOUT, TimeUnit.SECONDS);
+    } catch (InterruptedException ignore) {
+      interrupted = true;
     }
 
-    checkViewIsDisplayed(R.id.mapView);
-    Assert.assertNotNull(mapboxMap);
-  }
+    if (!interrupted) {
+      Timber.e("Timeout occurred for %s", testName.getMethodName());
 
-  protected MapboxMap getMapboxMap() {
-    return mapboxMap;
+      // update global timeout counter
+      timeoutCount++;
+      if (timeoutCount > MAX_TIMEOUT_FAILURE) {
+        // Too many timeouts, stop execution test runner
+        Timber.e("Too many timeouts occurred. Crashing Test-Suite.");
+        System.exit(-1);
+      }
+    }
   }
 
-  protected abstract Class getActivityClass();
-
-  protected void checkViewIsDisplayed(int id) {
-    onView(withId(id)).check(matches(isDisplayed()));
+  @Override
+  public void onDidFinishLoadingStyle() {
+    latch.countDown();
```

Comment:
How about we move this call from here to `OnStyleLoaded` of the `MapboxMap#getStyle(OnStyleLoadedListener)`? This would simplify the setup.